File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack, answer = [], 0
    for i in ingredient:
        stack.append(i)
        if len(stack) > 3 and stack[-4:] == [1, 2, 3, 1]:
            answer += 1
            for _ in range(4):
                stack.pop()
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    
    Stack<Integer> stack = new Stack<>();
    
    public int solution(int[] ingredient) {
        int answer = 0;
        for (int i : ingredient) {
            stack.add(i);
            int size = stack.size();
            if (size > 3 && check(size - 1)) {
                for (int k = 0; k < 4; k++) {
                    stack.pop();
                }
                answer++;
            }
        }
        return answer;
    }
    
    public boolean check(int idx) {
        if (stack.get(idx) != 1 || stack.get(idx - 3) != 1) return false;
        if (stack.get(idx - 1) != 3) return false;
        if (stack.get(idx - 2) != 2) return false;
        return true;
    }
}